### ISSUE TITLE
Document passing variables along with errors

### ISSUE THREAD

**Issue by korthout** [camunda/camunda-platform-docs#1510]
Camunda Platform `8.2.0-alpha2` will support passing variables along with errors thrown for jobs. We will need to document this feature.

We need to:
- Update the [gRPC API reference](https://docs.camunda.io/docs/next/apis-clients/grpc/#throwerror-rpc)
- Clarify the variable propagation behavior for [Error Events](https://docs.camunda.io/docs/next/components/modeler/bpmn/error-events/) (they are not propagated, i.e. they are always created on the local scope of the catching event, and output mappings can be used as always to control how the variables are merged into the process instance)

We may need to:
- Describe the use case in the [error handling concept](https://docs.camunda.io/docs/next/components/concepts/workflow-patterns/#handling-errors)

Introduced with:
- https://github.com/camunda/zeebe/pull/10882

**Comment by korthout** [comment]
@akeller It seems I no longer have the same access to this repo. I'm unable to label this issue. I'm also no longer able to assign it to the Zeebe Process Automation Team project.

**Comment by korthout** [comment]
Ok, seems I'm able to do do this again :D

**Comment by akeller** [comment]
IT was moving some permissions around and making an employee group which may have been right when you were trying to do this.

**Comment by menski** [comment]
Has to be done before 8.2 release
